crates/ra_ssr/src/tests.rs

```diff
@@ -512,6 +556,7 @@ fn preserves_whitespace_within_macro_expansion() {
 #[test]
 fn match_failure_reasons() {
     let code = r#"
+        fn bar() {}
         macro_rules! foo {
             ($a:expr) => {
                 1 + $a + 2
```

Comment:
I don't think this is directly applicable here, but we've recently introduced two new tools for testing:

* [inline annotations](https://github.com/rust-analyzer/rust-analyzer/commit/e805e8c1d5bf26e9716fb855f97d950395129c20#diff-60e4ae10c7f2ae1ed09a30452a3110d8)
* [snapshot testing library](https://github.com/rust-analyzer/rust-analyzer/pull/5101)

Reply:
Thanks, I'll look over. Good to know what testing facilities exist so that I can reuse them when applicable